### ISSUE TITLE
Discrepency on JSR/RTS handling

### ISSUE THREAD

**Issue by pweingar** [Trinity-11/FoenixIDE#10]
The emulator seems to be pushing the incorrect address to the stack on a JSR, compared to the hardware. JSR is supposed to push the address of the last byte of the JSR instruction. When RTS executes, it pulls the address off the stack and starts executing the _next_ byte. That is, if the address $1234 was pushed by JSR, RTS will start executing the opcode at $1235.

It appear that RTS in the emulator is starting execution at the address pushed on the stack ($1234) in the example above. JSR seems to be pushing a consistent value, so they work together.

Most code should be fine with this, but there will be some programs that push their own return values to the stack and use RTS to jump to them. These will fail on the emulator.

I'm attaching sample code that shows the discrepancy between the hardware and the emulator. The hardware will print "X" and break to the monitor. The emulator will just break to the monitor.

By the way, I have not tested if this is an issue with JSL/RTL.

```
;;;
;;; Example of issue with JSR/RTS
;;;

.cpu "65816"

FK_PUTC             = $190018 ; Print a character to the currently selected channel

;
; Bank 0 Code Segment, including bootup and variables
;

* = $001000

START       CLC
            XCE

            REP #$30
            .al
            .xl

            LDA #<>PROC         ; Push PROC as a return address.
            PHA                 ; Execution should continue at PROC+1
            RTS

lock        NOP
            BRA lock

PROC        BRK                 ; Emulator will execute this BRK
            
            SEP #$20            ; Hardware will skip the BRK and start here
            .as
            LDA #'X'
            JSL FK_PUTC         ; ... and print an "X"

            BRK


* = $00FFFC 

VRESET      .word <>START
```

**Comment by pweingar** [comment]
[rts_error.zip](https://github.com/Trinity-11/FoenixIDE/files/3649869/rts_error.zip)

**Comment by dtremblay** [comment]
I confirm that the emulator is indeed not pushing the correct address to the stack.  I'll have a fix shortly, I hope.

**Comment by dtremblay** [comment]
This issue was fixed in release 0.2.4.9 of the Foenix IDE.
